service/storage_service.cc

```diff
@@ -421,11 +421,11 @@ future<> storage_service::topology_state_load() {
                 // for the node were not in the 'normal' state yet
                 co_await update_peer_info(ip);
                 // And then amend with the info from raft
-                co_await _sys_ks.local().update_tokens(ip, rs.ring.value().tokens);
                 co_await _sys_ks.local().update_peer_info(ip, "data_center", rs.datacenter);
                 co_await _sys_ks.local().update_peer_info(ip, "rack", rs.rack);
-                co_await _sys_ks.local().update_peer_info(ip, "host_id", id.uuid());
                 co_await _sys_ks.local().update_peer_info(ip, "release_version", rs.release_version);
```

Comment:
Having many `INSERT`-s for a single row in `system.peers` seems wasteful. The `system.peers` table uses `schema_commitlog`, so each `INSERT` reaches the disk with `O_DSYNC`. We should probably
  * make a batch of info we want to write into `system.peers`;
  * compare it with the current state in the table;
  * write the changes in a single `INSERT`, if there are any. 

This is out of scope of this PR, deserving a separate one, WDYT @kbr-scylla

Reply:
makes sense to me and can be done as a follow up optimization

Reply:
> compare it with the current state in the table;

That would actually make it much more expensive, by introducing a read.

No need to do the comparison; we could keep doing unconditional write (but a single one instead of many).

Doesn't matter anyway. This is not a hot path.

Reply:
> That would actually make it much more expensive, by introducing a read.

Home come read (from cache, I suppose) can be more expensive than write with `O_DSYNC`?

> Doesn't matter anyway. This is not a hot path.

I wonder why, `O_DSYNC` write should be extremely expensive, IIUC.

Reply:
> Home come read (from cache, I suppose) can be more expensive than write with O_DSYNC?

Doesn't have to be from cache. If we ensured that it's cached then maybe.

> I wonder why, O_DSYNC write should be extremely expensive, IIUC.

By not a hot path I mean not used often. It only happens once per topology change. The hot paths are user data writes/reads which happen thousands times per second per shard.

Reply:
This is just unjustified pessimization, having them here and there makes the overall topology change procedure slow.

Reply:
done in v2